cmake/Findpybind11.cmake

```diff
@@ -2,18 +2,48 @@
 # See LICENSE.txt for details.
 
 find_path(
-    pybind11_INCLUDE_DIR
+    pybind11_INCLUDE_DIRS pybind11/pybind11.h
     PATH_SUFFIXES include
-    NAMES pybind11/pybind11.h
     HINTS ${pybind11_ROOT} ENV pybind11_ROOT
-    DOC "Pybind11 include directory. Used pybind11_ROOT to set a search dir."
+    DOC "Pybind11 include directory. Use pybind11_ROOT to set a search dir."
 )
 
-set(pybind11_INCLUDE_DIRS ${pybind11_INCLUDE_DIR})
+set(pybind11_INCLUDE_DIR ${pybind11_INCLUDE_DIRS})
```

Comment:
It seems like it would've helped Will had you checked that finding the dirs succeeded before parsing out version. Is there an easy way to do that?

Reply:
You could put in a check, but, actually, `/home/wthrowe/spectre/pybind11_INCLUDE_DIRS-NOTFOUND/pybind11/detail/common.h` was fairly clear.  (And in the other case it was found, it just found the wrong one.)

Reply:
Okay, fair enough. It's behind a check that someone explicitly required python bindings anyway